controllers/replica_set_controller.go

```diff
@@ -680,6 +678,11 @@ func (r ReplicaSetReconciler) buildAutomationConfig(mdb mdbv1.MongoDBCommunity)
 		}
 	}
 
+	if mdb.Spec.IsAgentX509() {
```

Comment:
What happens to the watchers if we remove x509 setting, do we remove them?

Reply:
Indeed the secrets will still be watched, but it will not affect anything (at most, when one of those secrets is added/deleted/updated, the reconciliation loop will run). But looking at the exisiting code, the watcher never removes any of the resources: prometheus, tls, user passwords. Do you think it's worth to add the functionality of "unwatching" resources?

Reply:
Yeah it should be fine, there are the following unlikely cases:
- memory leak in case of a lot of resources (highly unlikely)
- someone creates a secret with the same name which can cause mdb to reconcile even though its not related 

But it is nothing that you have to take care here 👍 . Can I ask you to create a jira issue for this?

Thanks!

Reply:
Yes I will!